zebra-rpc/src/methods/get_block_template_rpcs/get_block_template.rs

```diff
@@ -231,11 +238,15 @@ where
     Ok(chain_info)
 }
 
-/// Returns the transactions that are currently in `mempool`.
+/// Returns the transactions that are currently in `mempool`, or None if the
+/// `last_seen_tip_hash` from the mempool response doesn't match the tip hash from the state.
 ///
 /// You should call `check_synced_to_tip()` before calling this function.
 /// If the mempool is inactive because Zebra is not synced to the tip, returns no transactions.
-pub async fn fetch_mempool_transactions<Mempool>(mempool: Mempool) -> Result<Vec<VerifiedUnminedTx>>
+pub async fn fetch_mempool_transactions<Mempool>(
+    mempool: Mempool,
+    chain_tip_hash: block::Hash,
```

Comment:
Optional nitpick, I'm not sure if this is clearer or makes it more confusing!

```suggestion
    state_chain_tip_hash: block::Hash,
```